autoscheduler/frontend/src/components/SchedulingPage/SchedulePreview/ScheduleDetails/ScheduleDetails.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+import { useSelector } from 'react-redux';
+import {
+  Dialog, DialogContent, DialogTitle, Divider, IconButton, ThemeProvider, Typography,
+} from '@material-ui/core';
+import ChevronLeft from '@material-ui/icons/ChevronLeft';
+import ChevronRight from '@material-ui/icons/ChevronRight';
+import { RootState } from '../../../../redux/reducer';
+import Schedule from '../../../../types/Schedule';
+import * as styles from './ScheduleDetails.css';
+import sectionsForSchedule from '../../../../utils/sectionsForSchedule';
+import HonorsIcon from '../../../Icons/HonorsIcon/HonorsIcon';
+import Section from '../../../../types/Section';
+import GradeDist from '../../CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/SectionSelect/GradeDist/GradeDist';
+import meetingsForSection from '../../../../utils/meetingsForSection';
+import formatMeetingDays from '../../../../utils/formatMeetingDays';
+import meetingBuilding from '../../../../utils/meetingBuilding';
+import MeetingTypeDisplay from '../../CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/SectionSelect/MeetingType/MeetingTypeDisplay';
+import meetingTimeText from '../../../../utils/meetingTimeText';
+import CRNDisplay from './CRNDisplay/CRNDisplay';
+import InstructionalMethodIcon from '../../CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/SectionSelect/InstructionalMethodIcon/InstructionalMethodIcon';
+import { buttonTheme } from '../../../../theme';
+
+interface ScheduleDetailsProps {
+  open: boolean;
+  idx: number;
+  onClose: () => any;
+}
+
+const ScheduleDetails: React.FC<ScheduleDetailsProps> = ({
+  open: initialOpen, idx: initialIdx, onClose,
+}) => {
+  const schedules = useSelector<RootState, Schedule[]>((state) => state.schedules);
+  const [open, setOpen] = React.useState(initialOpen);
+  const [idx, setIdx] = React.useState(initialIdx);
+
+  // If parent updates state, apply those changes
+  React.useEffect(() => {
```

Comment:
You're correct that it should generally be avoided, and the react docs have a page about it [here](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html)

You'll see though that the material ui dialog itself has a similar implementation to what I did (using the `open` prop and a callback to update parent state):
![image](https://user-images.githubusercontent.com/28655462/105616659-3f1ccc80-5d9e-11eb-858c-4b07e2504f23.png)

Since the parent and children both need to be able to control whether the dialog is closed, the only real workaround I can think of would be to use redux. I wanted to keep the API for the dialog close to the vanilla `<Dialog />` component though, and adding a slice to redux only to keep track of whether the dialog is open seemed overkill to me